python/mxnet/ndarray/ndarray.py

```diff
@@ -83,6 +84,7 @@
     5: np.int8,
     6: np.int64,
     7: np.bool_,
+    12: np.dtype([('bfloat16', np.uint16)]),
```

Comment:
why 12?

Reply:
This is to align the `TypeFlag` defined in mshadow